### PR TITLE
CBL-138: fix: retry logic

### PR DESCRIPTION
**Pull request by jayahariv**
* previous retry was not cancelled, in case retry already scheduled.
* used the CBLTimer to schedule it.
* CBL-138
* #2525

### CODE REVIEW

File: Objective-C/CBLReplicator.mm [hunk 1]
```diff
@@ -165,17 +167,38 @@ - (void) start {
...
+
+- (void) scheduleRetry: (NSTimeInterval)delayInSeconds {
+    [CBLTimer cancel: _retryTimer];
+    
+    __weak CBLReplicator *weakSelf = self;
+    _retryTimer = [CBLTimer scheduleIn: _dispatchQueue
+                   after: delayInSeconds
+                   block:^{
+        [weakSelf _retry];
+    }];
```
Comment (pasin) [inline Objective-C/CBLReplicator.mm]: Please fix the indentation.

File: Objective-C/CBLReplicator.mm [hunk 2]
```diff
@@ -165,17 +167,38 @@ - (void) start {
...
+- (void) _retry {
     CBL_LOCK(self) {
-        CBLLogInfo(Sync, @"%@: Retrying...", self);
-        if (reset)
-            _retryCount = 0;
-        if (_repl || _state <= kCBLStateStopping) {
-            CBLLogInfo(Sync, @"%@: Ignore retrying (state = %d, status = %d)",
-                       self, _state, _rawStatus.level);
-            return;
+        if (_state == kCBLStateOffline || _state == kCBLStateSuspended) {
```
Comment (pasin) [inline Objective-C/CBLReplicator.mm]: Not sure if we need this if check is required. It seems like `if (_repl || _state <= kCBLStateStopping)` has already covered it.

Comment (jayahariv) [reply Objective-C/CBLReplicator.mm]: yes. you are right.

File: Objective-C/CBLReplicator.mm [hunk 3]
```diff
@@ -173,12 +173,12 @@ - (void) resetAndScheduleRetry {
...
 - (void) scheduleRetry: (NSTimeInterval)delayInSeconds {
-    [CBLTimer cancel: _retryTimer];
+    [self cancelRetry];
     
     __weak CBLReplicator *weakSelf = self;
     _retryTimer = [CBLTimer scheduleIn: _dispatchQueue
-                   after: delayInSeconds
-                   block:^{
+                                 after: delayInSeconds
+                                 block:^{
```
Comment (pasin) [inline Objective-C/CBLReplicator.mm]: Add a space between: and ^.

**Pull request merged**